### PR TITLE
CSP-670-Previous Engagement-CheckYourAnswers

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Web/Infrastructure/Category.cs
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Infrastructure/Category.cs:1]: Move this class under ...Domain.Constants folder

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/PreviousEngagementController.cs [hunk 1]
```diff
@@ -64,4 +56,14 @@ public IActionResult Post(PreviousEngagementSubmitModel submitmodel)
 
         return RedirectToRoute(RouteNames.Onboarding.CheckYourAnswers);
     }
+
+    private PreviousEngagementViewModel GetViewModel(OnboardingSessionModel sessionModel)
+    {
+        var previousEngagement = sessionModel.GetProfileValue(ProfileDataId.EngagedWithAPreviousAmbassadorInTheNetwork);
+        return new PreviousEngagementViewModel()
+        {
+            EngagedWithAPreviousAmbassadorInTheNetwork = bool.TryParse(previousEngagement, out var result) && result,
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/PreviousEngagementController.cs:65]: rename the `EngagedWithAPreviousAmbassadorInTheNetwork` to  `HasPreviousEngagement`

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs [hunk 2]
```diff
@@ -25,10 +25,11 @@ public class PreviousEngagementControllerPostTests
         [Frozen] PreviousEngagementSubmitModel submitmodel)
     {
         OnboardingSessionModel sessionModel = new();
-        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.Onboarding.AreasOfInterest);
-
+        sessionModel.ProfileData.Add(new ProfileModel { Id = ProfileDataId.EngagedWithAPreviousAmbassadorInTheNetwork, Value = "True" });
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs:28]: Rename the profile id to HasPreviousEngagement. The test is to check if there is no selection, so I think the Value should be left null here.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs:1]: There are multiple things in this file that needs addressing. 
SubmitModel on line 48 should not have [Frozen] attribute.

Update session model should test for both true and false values, currently it is only testing for true value. 

Line # 66 is unnecessary `sessionModel.ProfileData.Add(new ProfileModel { Id = ProfileDataId.EngagedWithAPreviousAmbassadorInTheNetwork, Value = "True" });`

There is no verify on session model being set from submit model in `UpdatesSessionModel` test.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerGetTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerGetTests.cs:1]: `Get_ViewModel_SelectedYes_RestoreFromSession` only covers case where value is true, case for false and null is missing.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs:1]: [Frozen] attribute on Line#48  is removed.

Update session model is set to string value, do you think it matters here to check for Boolean values? We are passing something and we are evaluating with the same value.

Line #66 is removed.

Added verify on session model, last two issues seems to be carried from last ticket. Thanks for spotting.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerGetTests.cs
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerGetTests.cs:1]: We do have another test named Get_ViewModel_SelectedNo_RestoreFromSession

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/WhenGetIsInvoked/AndSessionModelIsNotPopulated.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    public void ThenReturnsViewResults()
+    {
+        getResult.Should().NotBeNull();
+        getResult.ViewName.Should().Be(CheckYourAnswersController.ViewPath);
+    }
+
+    [Test]
+    public void ThenSetsJobTitleToNullInViewModel()
+    {
+        viewModel.JobTitle.Should().Be(JobTitle);
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/WhenGetIsInvoked/AndSessionModelIsNotPopulated.cs:55]: minor quibble, as there isn't really anything wrong with these tests, but Should().BeNull() in these instances make it a little more clear what is expected behaviour to anyone looking at these.   

Leave it as is, or update, your decision.

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs:1]: Fixed this

**Review by DevenKShah — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**